Domain.Authentication/Builders/SecurityTokenDescriptorBuilder.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Security.Claims;
+using Microsoft.IdentityModel.Tokens;
+
+namespace Diagnosea.Submarine.Domain.Authentication.Builders
+{
+    public class SecurityTokenDescriptorBuilder
+    {
+        private readonly IList<Claim> _claims = new List<Claim>();
+        private DateTime _expiration;
+        private SymmetricSecurityKey _symmetricSecurityKey;
+
+        public SecurityTokenDescriptorBuilder WithClaim(string type, string value)
+        {
+            _claims.Add(new Claim(type, value));
+            return this;
+        }
+        
+        public SecurityTokenDescriptorBuilder WithExpiration(DateTime expiration)
+        {
+            _expiration = expiration;
+            return this;
+        }
+        
+        public SecurityTokenDescriptorBuilder WithSymmetricSecurityKey(byte[] encodedSecret)
```

Comment:
Consider renaming to `WithEncodedSecret`. The fact its used to generate a security key is irrelevant to the implementor.